pkg/apis/core/validation/validation.go

```diff
@@ -5899,13 +5899,20 @@ func validateConfigMapNodeConfigSource(source *core.ConfigMapNodeConfigSource, f
 	return allErrs
 }
 
-// Validate compute resource typename.
+// Validate that the resource typename is qualified.
 // Refer to docs/design/resources.md for more details.
-func validateResourceName(value string, fldPath *field.Path) field.ErrorList {
+func validateResourceNameIsQualified(value string, fldPath *field.Path) field.ErrorList {
```

Comment:
This code used to be part of `validateResourceName`. The call chains used to be:

- `validateContainerCommon` -> `ValidateResourceRequirements` -> `validateContainerResourceName` -> `validateResourceName`
- `ValidateResourceQuota` -> `ValidateResourceQuotaSpec` -> `ValidateResourceQuotaResourceName` -> `validateResourceName`

Therefore all names valid in a `ResourceQuota` (second chain) also had to be valid in a container (first chain). That's probably why "persistentvolumeclaims" is valid in a container although it makes no sense there.

By splitting up `validateResourceName`, `ValidateResourceQuotaResourceName` can call `validateResourceNameIsQualified` and then consult it's own, separate list via `IsStandardQuotaResourceName` - that check was already there before, it just got combined with `IsStandardResourceName` unnecessarily.